src/App.js

```diff
@@ -1,26 +1,90 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { useState } from 'react'
 import './App.css';
+import contacts from './contacts.json'
+
+const App = () => {
+  //Iteration 1 | Display 5 Contacts
+  const fiveContacts = contacts.slice(0, 5);
+  const [contactsState, setcontactsState] = useState(fiveContacts);
+
+  //Iteration 2 | Add New Random Contacts
+  const addRandomContacts = () => {
+    const randomContacts = Math.floor(Math.random() * contacts.length);
+    const contactsStateCopy = [...contactsState];
+    contactsStateCopy.push(contacts[randomContacts]);
+    setcontactsState(contactsStateCopy);
+  }
+
+  //Iteration 3 a | Sort Contacts By Name
+  const sortByName = () => {
+    const contactsStateCopy = [...contactsState];
+
+    contactsStateCopy.sort(function (a, b) {
+      if (a.name < b.name) { return -1; }
+      if (a.name > b.name) { return 1; }
+      return 0;
+    });
+    setcontactsState(contactsStateCopy);
+  }
+
+  //Iteration 3 b | Sort Contacts By Popularity
+  const sortByPopularity = () => {
+    const contactsStateCopy = [...contactsState];
+
+    contactsStateCopy.sort(function (a, b) {
+      if (a.popularity < b.popularity)
+        return -1;
+      if (a.popularity > b.popularity)
+        return 1;
+      return 0;
+
+    });
+    setcontactsState(contactsStateCopy);
+  }
+
+  //Iteration 4 | Remove Contacts
+  const removeContacts = (id) => {
+    const contactsStateCopy = [...contactsState];
+
+    const index = contactsStateCopy.findIndex((contact) => contact.id === id);
+    contactsStateCopy.splice(index, 1);
+    setcontactsState(contactsStateCopy);
+  }
 
-function App() {
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <h1>IronContacts</h1>
+
+      <div className="buttonsContainer">
+        <button onClick={() => addRandomContacts()}>Add Random Contact</button>
+        <button onClick={() => sortByName()}>Sort by name</button>
+        <button onClick={() => sortByPopularity()}>Sort by popularity</button>
+      </div>
+
+      <div className="tableofContacts">
+        <table>
+          <tr>
+            <th>Picture</th>
+            <th>Name</th>
+            <th>Popularity</th>
+            <th>Action</th>
+          </tr>
+          {contactsState.map(item => (
+
+            <tr>
```

Comment:
Please give this `tr` a key property with the id as the value, as required by React (check the console).